File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        vector<int> result;
        queue<TreeNode*> nodeque;
        if (root != nullptr) nodeque.push(root);
        while (! nodeque.empty()){
            int size = nodeque.size();
            int level_max = INT_MIN;
            for (int i = 0; i < size; i++){
                TreeNode* cur = nodeque.front();
                nodeque.pop();
                level_max = cur->val > level_max? cur->val : level_max;
                if (cur->left) nodeque.push(cur->left);
                if (cur->right) nodeque.push(cur->right);
            }            
            result.push_back(level_max);
        }
        return result;
        }
};

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def largestValues(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        result = []
        if root == None:
            return []
        def recurse(node, level):
            if node == None:
                return
            if len(result) == level:
                result.append(float('-inf'))
            result[level] = max(result[level], node.val)
            if node.left:
                recurse(node.left, level+1)
            if node.right:
                recurse(node.right, level+1)
        recurse(root, 0)
        return result